deepreg/model/loss/label.py

```diff
@@ -323,89 +303,3 @@ def get_config(self):
 @REGISTRY.register_loss(name="jaccard")
 class JaccardLoss(NegativeLossMixin, JaccardIndex):
     """Revert the sign of JaccardIndex."""
-
-
-def separable_filter(tensor: tf.Tensor, kernel: tf.Tensor) -> tf.Tensor:
-    """
-    Create a 3d separable filter.
-
-    Here `tf.nn.conv3d` accepts the `filters` argument of shape
-    (filter_depth, filter_height, filter_width, in_channels, out_channels),
-    where the first axis of `filters` is the depth not batch,
-    and the input to `tf.nn.conv3d` is of shape
-    (batch, in_depth, in_height, in_width, in_channels).
-
-    :param tensor: shape = (batch, dim1, dim2, dim3)
-    :param kernel: shape = (dim4,)
-    :return: shape = (batch, dim1, dim2, dim3)
-    """
-    strides = [1, 1, 1, 1, 1]
-    kernel = tf.cast(kernel, dtype=tensor.dtype)
-    tensor = tf.nn.conv3d(
-        tf.nn.conv3d(
-            tf.nn.conv3d(
-                tf.expand_dims(tensor, axis=4),
-                filters=tf.reshape(kernel, [-1, 1, 1, 1, 1]),
-                strides=strides,
-                padding="SAME",
-            ),
-            filters=tf.reshape(kernel, [1, -1, 1, 1, 1]),
-            strides=strides,
-            padding="SAME",
-        ),
-        filters=tf.reshape(kernel, [1, 1, -1, 1, 1]),
-        strides=strides,
-        padding="SAME",
-    )
-    return tensor[:, :, :, :, 0]
-
-
-def compute_centroid(mask: tf.Tensor, grid: tf.Tensor) -> tf.Tensor:
```

Comment:
did i miss anything? where are the three functions now? compute_centroid, compute_centroid_distance, foreground_proportion

Reply:
They did not seem to be used in label.py and I thought it was some left-overs from a refactorization. I'm copying them to loss/util.py

Reply:
oh, now I see. I think that they should be refactored as classes just as like the other metrics in the file. Moreover, we may want to create a separate metrics.py as losses and metrics do not always match (e.g. dice)